sockets/sockets.js

```diff
@@ -320,7 +322,57 @@ var actionsObj = {
 					return { message: (Math.floor(Math.random() * 10) + 1).toString(), classStr: "server-text" }
 				}
 			}
-		},
+        },
+        
+        mods: {
+            command: 'mods',
+            help: '/mods: Shows a list of online moderators.',
+            run() {
+                const modUsers = getPlayerUsernamesFromAllSockets().filter((username) => modsArray.includes(username));
+                const message = `Currently online mods: ${modUsers.length > 0 ? modUsers.join(', ') : 'None'}.`;
+                return { message, classStr: "server-text" };
+            }
+        },
+
+        pmmod: {
+            command: 'pmmod',
+            help: '/pmmod: Sends a private message to an online moderator.',
+            run(data, senderSocket) {
+                const { args } = data;
+                // We check if they are spamming, i.e. have sent a PM before the timeout is up
+                const lastPmTime = pmmodCooldowns[senderSocket.id];
+                if (lastPmTime) {
+                    const remaining = new Date() - lastPmTime;
+                    if (remaining < PMMOD_TIMEOUT) return { message: `Please wait ${Math.ceil((PMMOD_TIMEOUT - remaining) / 1000)} seconds before sending another pm!` };
+                }
+                // Checks for various missing fields or errors
+                if (!args[1]) return { message: 'Please specify a mod to message. Type /mods to get a list of online mods.', classStr: 'server-text' };
+                if (!args[2]) return { message: 'Please specify a message to send.', classStr: 'server-text' };
+                if (!modsArray.includes(args[1])) return { message: 'That user is not a mod. You may not private message them.', classStr: 'server-text' };
+                const modSocket = allSockets[getIndexFromUsername(allSockets, args[1], true)];
+                if (!modSocket) return senderSocket.emit('messageCommandReturnStr', { message: "Could not find " + args[1], classStr: "server-text" });
+
+				let str = `${senderSocket.request.user.username}->${args[1]} (whisper): ${args.slice(2).join(' ')}`;
+                
+				const dataMessage = {
+					message: str,
+					dateCreated: new Date(),
+					classStr: "whisper"
+                };
+                
+                senderSocket.emit('allChatToClient', dataMessage);
+                senderSocket.emit('roomChatToClient', dataMessage);
+
+                modSocket.emit('allChatToClient', dataMessage);
+                modSocket.emit('roomChatToClient', dataMessage)
+
+                lastWhisperObj[modSocket.request.user.username] = senderSocket.request.user.username;
```

Comment:
Will this allow the players to abuse the commands?

Player does /pmmod <mod> <message>. 
Player does /r <message> to respond to himself. They can then spam the /r command. (Sorry I haven't tested this yet, this is just a code review)

Reply:
Ok I'll just add a bit so that they can't `/pmmod` themselves? Would that work?

Reply:
I'm not sure if it will cause a problem, but I think when I wrote the /r command, it checks if the user exists in the lastWhisperObj. If they do, then they can use /r. So if you included it here, even if a moderator doesn't respond to them, the player abusing this command can then spam /r messages to mods? I haven't verified that this issue exists.